### ISSUE TITLE
The model is using the test values decomposition as covariates

### ISSUE THREAD

**Issue by qlzfelipe** [cran/WaveletSVR#1]
Hello,

There seems to be a serious problem with the way `WaveletFittingsvr() ` is being implemented.

You are using the full dataset (including the test set) to create the wavelet covariates, instead of having the forecasts happen in a recursive manner using the lagged value from the last observation in the training set.

This can be seen in line 99, where you are using the test values' decomposition as covariates for generating the forecasts, which will, of course, produce good but invalid results. This creates the assumption the test set is already know, which is not true in an environment where you want to forecast unknown future values.

I first suspected that was happening because the function uses the full dataset instead of the only the training set, which is hardly appropriate if you look into most model training functions' standards.

Then I ran some tests like the one below:

```{r}
library(WaveletSVR)
set.seed(123)

data <- rnorm(100)
#Running with the original dataset
WaveletFittingsvr(data, Waveletlevels = 2, SplitRatio = 0.8)

#Running with the original training dataset, replacing test values with something else.
test <- rnorm(20)
data[81:100] <- test

WaveletFittingsvr(data, Waveletlevels = 2, SplitRatio = 0.8)
```
You can verify the results are different.

Shouldn't they be the same, given the training dataset is the same?

And by the way, why they output 18 and 19 forecasts given the test set has 20 observations? It seems there are multiple issues with this function.

Please let me know if the issue description isn't clear enough.

**Comment by gaborcsardi** [comment]
Hi, this is a read only mirror of CRAN, please see the package authors in the `DESCRIPTION` file. Look for `Maintainer`, `BugReports` and `URL`. Thanks!
